redis/redis.go

```diff
@@ -0,0 +1,173 @@
+package redis
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"github.com/go-redis/redis"
+	"sync"
+	"time"
+
+	"github.com/cncd/queue"
+)
+
+const POP_TIMEOUT = 0 // 0 == Blocking forever
+
+var ErrDeadLine = errors.New("queue: deadline received")
+
+type entry struct {
+	item     *queue.Task
+	done     chan bool
+	retry    int
+	error    error
+	deadline time.Time
+}
+
+type conn struct {
+	sync.Mutex
+
+	opts      *Options
+	client    *redis.Client
+	running   map[string]*entry
```

Comment:
I think we should store this information in redis so that we can restart the redis server or drone server without losing the state of the queue.